app/src/test/java/com/codesoom/assignment/infra/UserRepositoryTest.java

```diff
@@ -0,0 +1,49 @@
+package com.codesoom.assignment.infra;
+
+import static com.codesoom.assignment.domain.UserConstants.USER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+@DataJpaTest
+@DisplayName("UserRepository 클래스")
+public class UserRepositoryTest {
+    @Autowired
+    UserRepository userRepository;
+
+    private User savedUser;
+
+    void subjectSave() {
+        savedUser = userRepository.save(USER);
+    }
+
+    void subjectDelete() {
+        userRepository.delete(savedUser);
+    }
+
+    @Nested
+    @DisplayName("save 메서드는")
+    class Context_user_save {
+        @AfterEach
+        void afterEach() {
+            subjectDelete();
+        }
+
+        @Test
+        @DisplayName("User를 저장한다.")
+        void it_saves_object() {
+            subjectSave();
```

Comment:
이렇게 받은 다음, 결과를 검증해보세요.
```suggestion
            User saveResult = subjectSave();
```

Reply:
넵 수정하겠습니다.

Reply:
```
void subjectDelete() {
    userRepository.delete(savedUser);
}
```
```
@AfterEach
void afterEach() {
         subjectDelete();
}
```
이 부분 때문에
```
void subjectSave() {
        savedUser = userRepository.save(USER);
}
```
```
@Test
@DisplayName("User를 저장한다.")
void it_saves_object() {
         subjectSave();
         assertThat(savedUser)
                  .matches(user -> user.getId() != null);
}
```
이런식으로 작성하였습니다.

Reply:
subject 가 테스트하려는 메소드가 리턴값이 있다면, subject도 그것을 리턴하게 해 주세요. subject는 테스트 대상을 아주 얇게 감싼 것이어야 좋습니다. 리턴하는 것을 틀어막으면 좋지 않습니다.

Reply:
그렇군요! 수정하겠습니다.